projects/go-lib/src/lib/components/go-search/go-search.component.ts

```diff
@@ -0,0 +1,82 @@
+import { Component, ElementRef, HostListener, OnInit } from '@angular/core';
+import { AnimationEvent } from '@angular/animations';
+import { FormBuilder, FormGroup } from '@angular/forms';
+import { debounceTime, distinctUntilChanged } from 'rxjs/operators';
+
+import { searchLoaderAnim, searchResultsAnim } from '../../animations/search.animation';
+import { GoSearchService } from './go-search.service';
+
+@Component({
+  selector: 'go-search',
+  templateUrl: './go-search.component.html',
+  styleUrls: ['./go-search.component.scss'],
+  animations: [searchLoaderAnim, searchResultsAnim]
+})
+export class GoSearchComponent implements OnInit {
+
+  goSearchForm: FormGroup;
+  searchActive: boolean = false;
+  resultsOverflow: string = 'hidden';
+
+  @HostListener('document:click') onDocumentClick(event) {
+    this.closeSearchEvent(event);
+  }
+
+  constructor(
+    public goSearchService: GoSearchService,
+    private elementRef: ElementRef,
+    private fb: FormBuilder
```

Comment:
Do we think calling this `fb` would be something that is easy to guess if this component got a little more lengthy and the variable was referenced further away from where it is declared?

I personally looked at line 30 and was like WTF is `fb`.

Reply:
That's fair, I think it's common in Angular to name it that... but I've changed it as it doesn't hurt to be slightly more direct.